index.html

```diff
@@ -89,46 +141,199 @@
+      }); 
     }
 
     // SCATTERPLOT
+  
+
+    var points = chart.svg.selectAll('.point')
+                    .data(txData);
+
+                    points
+                    // This is what I would think would make them fade out but
+                    // it just breaks the code and nothing fades in:
+                    //.transition().duration(1000)
+                    .attr('r', 0);
+
+                    points.enter().append('circle')
+                    .attr('class', 'point')
+                    .attr('r', 0);
+
+                    points
+                    .attr('cx', function(d) { return chart.x(d.democratContribution); })
+                    .attr('cy', function(d) { return chart.y(d.republicanContribution)})
+                    .transition().duration(2000).attr('r', 5);
+
+                    points.exit()
+                    //.transition().duration(1000).attr('r',0)
+                    .remove();
 
     // HISTOGRAMS
 
+    var barWidth = ( chart.x(chart.x.ticks()[1]) - chart.x(chart.x.ticks()[0]) )/5;
+
+    // x axis histogram
+    var demContr = txData.map(function (i) {return i.democratContribution});
+
+    var xHistogram = d3.layout.histogram()
+                             .bins(chart.x.ticks())
+                             (demContr); 
+
+    histScale = d3.scale.linear()
+                    .domain([0, d3.max(xHistogram, function (d) { return d.y; })])
+                    .range([0, height/5])
+                    .nice();
+    
+    
+
+    var xBars = chart.svg.selectAll(".xbar")
+                  .data(xHistogram)
+                  ;
+    
+    xBars.enter().append("rect")
+                 .attr("class", "xbar")
+                 .attr("transform", function(d, i) { 
+                  return "translate(" 
+                       + ( ((chart.x(chart.x.ticks()[i+1]) - (chart.x(chart.x.ticks()[i])))/2) + chart.x(chart.x.ticks()[i]) - (barWidth/2) )  
+                       + ",0)"; 
+                     })
+                 .attr("y", height)
+                 .attr("width", barWidth)
+                 .attr("height", 0);
+
+    xBars
+       .transition().duration(2000)
+       .attr("y", function (d) { return height -  histScale(d.y)})
+       .attr("height", function (d) { return histScale(d.y)});
+
+   // y axis histogram
+    var repContr = txData.map(function (i) {return i.republicanContribution});
+
+    var yHistogram = d3.layout.histogram()
+                             .bins(chart.y.ticks())
+                             (repContr);                        
+
+    var yBars = chart.svg.selectAll(".ybar")
+                  .data(yHistogram);
+
+   yBars.enter().append("rect")
+        .attr("class", "ybar")
+        .attr("transform", function(d, i) { return "rotate (-90) translate(-" + ( ((chart.y(chart.y.ticks()[i+1]) - (chart.y(chart.y.ticks()[i])))/2) + chart.y(chart.y.ticks()[i]) + (barWidth/2) )  + ",0)"; })
+        .attr("width", barWidth)
+        .attr("height", 0);
+
```

Comment:
This is a really interesting way to do this, and it looks like it worked out for you. Note that you could have created a new group, used transform to rotate it, then appends rects within that group and used the (slightly easier syntax) for manipulating their x,y,width,height. Might be a bit more readable that way, but this seems to be working fine.